h2o-extensions/xgboost/src/main/java/hex/tree/xgboost/XGBoost.java

```diff
@@ -396,6 +396,24 @@ final void buildModelImpl() {
         model.unlock(_job);
       }
     }
+    
+    @Override
+    public void computeEffectiveParameters() {
```

Comment:
why is this not done in model.initEffectiveParams? overriding this method seems like a not good idea, ideally I would make the super method final, becuase overriding is dangerous (might forget to add checkEffectiveParmsDoesNotContainAuto for example)

Reply:
this is done here because I have here isTrainDatasetSparse() and train().numRows()

Reply:
moved to model